Sources/EdgeNetworkHandlers/RequestBuilder.swift

```diff
@@ -101,43 +103,68 @@ class RequestBuilder {
 
         for event in events {
             guard var eventData = event.data else {
                 continue
             }
 
-            if eventData[EdgeConstants.JsonKeys.XDM] == nil {
-                eventData[EdgeConstants.JsonKeys.XDM] = [:]
+            var xdm = eventData[EdgeConstants.JsonKeys.XDM] as? [String: Any] ?? [:]
+
+            if xdm[EdgeConstants.JsonKeys.TIMESTAMP] == nil ||
+                (xdm[EdgeConstants.JsonKeys.TIMESTAMP] as? String)?.isEmpty ?? true {
+                // if no timestamp is provided in the xdm event payload, set the event timestamp
+                xdm[EdgeConstants.JsonKeys.TIMESTAMP] = event.timestamp.getISO8601UTCDateWithMilliseconds()
             }
 
-            if var xdm = eventData[EdgeConstants.JsonKeys.XDM] as? [String: Any] {
+            xdm[EdgeConstants.JsonKeys.EVENT_ID] = event.id.uuidString
+            eventData[EdgeConstants.JsonKeys.XDM] = xdm
+
+            var metadata = [String: Any]()
+            // check if config overrides are set
+            if let eventConfigData = eventData[EdgeConstants.EventDataKeys.Config.KEY] as? [String: Any] {
+                // send original datastreamId in meta if override is set
+                if let datastreamIdOverride =
+                    eventConfigData[EdgeConstants.EventDataKeys.Config.DATASTREAM_ID_OVERRIDE] as? String {
+                    let trimmedDatastreamIdOverride = datastreamIdOverride.trimmingCharacters(in: CharacterSet.whitespaces)
 
-                if xdm[EdgeConstants.JsonKeys.TIMESTAMP] == nil ||
-                    (xdm[EdgeConstants.JsonKeys.TIMESTAMP] as? String)?.isEmpty ?? true {
-                    // if no timestamp is provided in the xdm event payload, set the event timestamp
-                    xdm[EdgeConstants.JsonKeys.TIMESTAMP] = event.timestamp.getISO8601UTCDateWithMilliseconds()
+                    let originalDatastream = config[EdgeConstants.SharedState.Configuration.CONFIG_ID]
+
+                    if !trimmedDatastreamIdOverride.isEmpty && trimmedDatastreamIdOverride != originalDatastream {
+                        metadata[EdgeConstants.JsonKeys.ConfigMetadata.SDK_CONFIG] = [EdgeConstants.JsonKeys.ConfigMetadata.DATASTREAM:
+                                    [EdgeConstants.JsonKeys.ConfigMetadata.ORGIGINAL: originalDatastream]]
+                    }
                 }
 
-                xdm[EdgeConstants.JsonKeys.EVENT_ID] = event.id.uuidString
-                eventData[EdgeConstants.JsonKeys.XDM] = xdm
+                // enable datastream config override if set
+                if let datastreamConfigOverrides = eventConfigData[EdgeConstants.EventDataKeys.Config.DATASTREAM_CONFIG_OVERRIDE] as? [String: Any] {
+                    if !datastreamConfigOverrides.isEmpty {
+                        metadata[EdgeConstants.JsonKeys.ConfigMetadata.CONFIG_OVERRIDES] = datastreamConfigOverrides
+                    }
+                }
```

Comment:
SDKConfig should be part of the top level RequestMetadata, not per event. The event metadata will remain unchanged, containing just dataset id when used.